app/models/comment.rb

```diff
@@ -647,83 +627,30 @@ def self.recent_threads(user)
       .pluck(:thread_id)
     return Comment.none if thread_ids.empty?
 
-    Comment
-      .joins(<<~SQL
-        inner join (
-          with recursive discussion as (
-          select
-            c.id,
-            0 as depth,
-            (select count(*) from comments where parent_comment_id = c.id) as reply_count,
-            cast(confidence_order as char(#{Comment::COP_LENGTH}) character set binary) as confidence_order_path
-            from comments c
-            where
-              thread_id in (#{thread_ids.join(", ")}) and
-              parent_comment_id is null
-          union all
-          select
-            c.id,
-            discussion.depth + 1,
-            (select count(*) from comments where parent_comment_id = c.id),
-            cast(concat(
-              left(discussion.confidence_order_path, 3 * (depth + 1)),
-              c.confidence_order
-            ) as char(#{Comment::COP_LENGTH}) character set binary)
-          from comments c join discussion on c.parent_comment_id = discussion.id
-          )
-          select * from discussion as comments
-        ) as comments_recursive on comments.id = comments_recursive.id
-      SQL
-            )
-      .order("comments.thread_id desc, comments_recursive.confidence_order_path")
-      .select('
-        comments.*,
-        comments_recursive.depth as depth,
-        comments_recursive.reply_count as reply_count
-      ')
+    # Going from top to botton to find all children in the thread
+    Comment.with_recursive(
+      comments_recursive: [
+        Comment.where(thread_id: thread_ids, parent_comment_id: nil),
+        Comment.joins("JOIN comments_recursive ON comments_recursive.id = comments.parent_comment_id")
+      ]
+    ).from("comments_recursive AS comments").order(thread_id: :desc)
```

Comment:
It will be nice to simplify the definition of `confidence_order_path` from `confidence_order`.